src/app/Rules/CreateRule.tsx

```diff
@@ -0,0 +1,374 @@
+    setPreservedArchives(Number(evt));
+  };
+
+  const handleSubmit = (): void => {
+    const notificationMessages: string[] = [];
+    if (nameValid !== ValidatedOptions.success) {
+      notificationMessages.push(`Rule name ${name} is invalid`);
+    }
+    if (notificationMessages.length > 0) {
+      const message = notificationMessages.join('. ').trim() + '.';
+      notifications.warning('Invalid form data', message);
+      return;
+    }
+    const rule: Rule = {
+      name,
+      description,
+      matchExpression,
+      eventSpecifier: getEventString(),
+      archivalPeriodSeconds: archivalPeriod * archivalPeriodUnits,
+      preservedArchives,
+      maxAgeSeconds: maxAge * maxAgeUnits,
+      maxSizeBytes: maxSize * maxSizeUnits
+    };
+    addSubscription(
+      context.api.createRule(rule)
+      .pipe(first())
+      .subscribe(success => {
+        if (success) {
+          history.push('/rules');
+        }
+      })
+    );
+  };
+
+  // FIXME we query ourselves to populate the list of templates, since Rules can apply to any target. Is this better than making the user write the event specifier manually,
+  // or at least make them write the name manually and choose TARGET/CUSTOM from a dropdown?
+  React.useEffect(() => {
+    addSubscription(
+      context.api.doGet<EventTemplate[]>(`targets/localhost:0/templates`).subscribe(setTemplates)
+    );
+  }, []);
+
+  const breadcrumbs: BreadcrumbTrail[] = [
+    {
+      title: 'Automated Rules',
+      path: '/rules',
+    }
+  ];
+
+  const templatesOptionGroups = React.useMemo(() => {
+    const offset = templates.filter(t => t.type === "CUSTOM").length;
+    return (<>
+      <FormSelectOption key="-1" value="" label="Select a Template" isPlaceholder />
+      <FormSelectOptionGroup key="-2" label="Custom Templates">
+        {
+          templates.filter(t => t.type === "CUSTOM").map((t: EventTemplate, idx: number) => (<FormSelectOption key={idx} value={`${t.name},${t.type}`} label={t.name} />))
+        }
+      </FormSelectOptionGroup>
+      <FormSelectOptionGroup key="-3" label="Target Templates">
+        {
+          templates.filter(t => t.type === "TARGET").map((t: EventTemplate, idx: number) => (<FormSelectOption key={idx+offset} value={`${t.name},${t.type}`} label={t.name} />))
+        }
+      </FormSelectOptionGroup>
+    </>);
+  }, [templates]);
+
+  return (
+    <BreadcrumbPage pageTitle='Create' breadcrumbs={breadcrumbs} >
+      <Card>
+        <CardBody>
+          <Form isHorizontal>
+            <Text component={TextVariants.small}>
+              Automated Rules are configurations that instruct Cryostat to create JDK Flight Recordings on matching target JVM applications
+              with specified parameters for which Event Template to use, how much data should be kept in the application recording buffer, and
+              how frequently Cryostat should copy the application recording buffer into Cryostat's own archived storage.
```

Comment:
Could you split this sentence into two sentences for clarity? Maybe something like "Automated Rules are configurations that instruct Cryostat to create JDK Flight Recordings on matching target JVM applications. Each Automated Rule specifies parameters for which Event Template to use, how much data should be kept in the application recording buffer, and how frequently Cryostat should copy the application recording buffer into Cryostat's own archived storage."